server/dals/protein-dal.js

```diff
@@ -1,56 +1,64 @@
-    return "SELECT * FROM protein_protein_interactions.source ORDER BY time_stamp DESC;";
+    return `SELECT * FROM protein_protein_interactions.source
+            UNION ALL
+            SELECT * FROM protein_protein_interactions_new.source
+            ORDER BY time_stamp DESC;`;
 };
 
-const buildNetworkFromGeneProteinQuery = function (geneProtein) {
+const buildNetworkFromGeneProteinQuery = function (geneProtein, source, timestamp) {
+    const namespace = timestampNamespace(timestamp, false);
+    const timestampQuery = isTimestampOld(timestamp)
+            ? ""
+            : `AND gene.time_stamp='${timestamp}' AND gene.source='${source}'`;
+    console.log("timestampquery", timestampQuery);
     return `SELECT DISTINCT gene_id, display_gene_id, standard_name, length, molecular_weight, PI FROM
- protein_protein_interactions.gene, protein_protein_interactions.protein WHERE
- (LOWER(gene.gene_id)=LOWER('${geneProtein}') OR LOWER(gene.display_gene_id)=LOWER('${geneProtein}') 
- OR LOWER(protein.standard_name) =LOWER('${geneProtein}')) AND
- LOWER(gene.gene_id) = LOWER(protein.gene_systematic_name);`;
+    ${namespace}.gene, ${namespace}.protein WHERE
+    (LOWER(gene.gene_id)=LOWER('${geneProtein}') OR LOWER(gene.display_gene_id)=LOWER('${geneProtein}')
+    OR LOWER(protein.standard_name) =LOWER('${geneProtein}')) AND
+    LOWER(gene.gene_id) = LOWER(protein.gene_systematic_name)${timestampQuery};`;
+
+    // do I need to check for source each time I check for timestamp?
 };
 
 const buildNetworkProteinsQuery = function (proteinString) {
     let proteins = "(";
     let proteinList = proteinString.split(",");
-    proteinList.forEach(x => proteins += ( `(physical_interactions.protein1 =\'${x}\') OR `));
+    proteinList.forEach((x) => (proteins += `(physical_interactions.protein1 =\'${x}\') OR `));
     proteins = `${proteins.substring(0, proteins.length - 4)}) AND (`;
-    proteinList.forEach(x => proteins += ( `(physical_interactions.protein2 =\'${x}\') OR `));
+    proteinList.forEach((x) => (proteins += `(physical_interactions.protein2 =\'${x}\') OR `));
     return `${proteins.substring(0, proteins.length - 4)})`;
-
 };
 
 const buildGenerateProteinNetworkQuery = function (proteins, timestamp, source) {
-    return `SELECT DISTINCT protein1, protein2 FROM
- protein_protein_interactions.physical_interactions WHERE
- physical_interactions.time_stamp='${timestamp}' AND physical_interactions.source='${source}' AND
- ${buildNetworkProteinsQuery(proteins)} ORDER BY protein1 DESC;`;
+    const namespace = timestampNamespace(timestamp, false);
+    const annotation = isTimestampOld(timestamp) ? "" : ", annotation_type";
+    return `SELECT DISTINCT protein1, protein2${annotation} FROM
+        ${namespace}.physical_interactions WHERE
+        physical_interactions.time_stamp='${timestamp}' AND physical_interactions.source='${source}' AND
+        ${buildNetworkProteinsQuery(proteins)} ORDER BY protein1 DESC;`;
```

Comment:
Just noting that the approach leaves us theoretically vulnerable to SQL injection attacks. For future work, we should look into safer ways to build our SQL queries